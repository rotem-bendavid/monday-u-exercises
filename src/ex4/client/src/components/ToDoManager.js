import React, { useState,useEffect,useCallback } from 'react';
import ToDoAdd from './ToDoAdd';
import ToDoList from './ToDoList';
import ToDoHide from './ToDoHide';
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import {getTodosAction} from '../redux/actions/todo-actions';

function ToDoManager(props) {
    const [loading, setLoading] = useState(false);

    async function importTodos() {
        try {
            await props.get_todos();
            setLoading(false);
        }
        catch{
            console.error("err: couldn't import todos");
            setLoading(true);
        }
    }

    useEffect(() => {
        importTodos();
    },[])

    useEffect(() => {
        if (loading) importTodos();
    },[loading])

    if (loading) return( <div>Loading..</div>);
    
    return (
    <>
        <div className="app-container">
            <div className="list-container">
                <h1>TO-DO list</h1>
                    <ToDoAdd/>
                    <ToDoHide/>
                    <ul id="list">
                        <ToDoList/>
                    </ul>
                    
            </div>
        </div>

    </>
    );
}

const mapStateToProps = (state) => {
    return {
      todos: state.itemsEntities.todos,
    };
  };

  const mapDispatchToProps = (dispatch) => ({
    get_todos: () => dispatch(getTodosAction()),
  });

export default connect(mapStateToProps,mapDispatchToProps)(ToDoManager);