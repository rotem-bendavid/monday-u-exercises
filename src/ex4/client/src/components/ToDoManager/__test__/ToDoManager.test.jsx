import ToDoManager from "../ToDoManager";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import renderer from "react-test-renderer";

test("shouldn't show listitem", () => {
  const tree = renderer.create(
      <Provider store={store}>
      <ToDoManager />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});