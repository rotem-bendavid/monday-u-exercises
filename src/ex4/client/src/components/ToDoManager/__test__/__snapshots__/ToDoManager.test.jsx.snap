// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shouldn't show listitem 1`] = `
<div
  className="app-container"
>
  <div
    className="list-container"
  >
    <h1>
      TO-DO list
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="new-task"
      >
        <input
          id="list-item-input"
          name="text"
          onChange={[Function]}
          placeholder="Add new TO-DO"
          value=""
        />
        <button
          id="list-item-submit"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </form>
    <div>
      <div
        className="todo-hide-buttons"
      >
        <button
          className="hide-bt-all"
          onClick={[Function]}
        >
          View all
        </button>
        <button
          className="hide-bt-active"
          onClick={[Function]}
        >
          Active
        </button>
        <button
          className="hide-bt-completed"
          onClick={[Function]}
        >
          Completed
        </button>
        <button
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <ul
      id="list"
    />
  </div>
</div>
`;
